internal/audit/agent.go

```diff
@@ -21,14 +21,22 @@ type Labels = map[string]any
 type Agent interface {
 	Trace(context.Context, Audit)
 	Cache() AuditCache
-	SetGlobalLabels(labels Labels)
+}
+
+type AgentPool interface {
+	New() Agent
+	SetGlobalLabels(Labels)
```

Comment:
I'm not convinced about this method being part of the pool behaviour. 

The method is intended to set labels during policy evaluation on a specific agent, if such labels are shared between multiple agents we might still have concurrency issues. 

Moreover I don't see changes to the builtin rego func that is using the method which feels odd, I'm worried about some type assertion that is making the code compile but that won't work at runtime